lib/loadGltfUris.js

```diff
@@ -0,0 +1,61 @@
+'use strict';
+var fs = require('fs');
+var Cesium = require('cesium');
+var defined = Cesium.defined;
+var dataUriToBuffer = require('data-uri-to-buffer');
+
+module.exports = loadGltfUris;
+
+function loadGltfUris(path, gltf, callback) {
+    var shaders = gltf.shaders;
+    if (defined(shaders)) {
+        for (var id in shaders) {
+            if (shaders.hasOwnProperty(id)) {
+                var shader = shaders[id];
+                shader.extras = shader.extras || {};
+
+                var uri = shader.uri;
+                loadURI(gltf, path, shader.extras, uri, callback);
+            }
+        }
+    }
+
+    return gltf;
+}
+
+function loadURI(gltf, path, extras, uri, callback) {
+    if (!/^data\:/i.test(uri)) {
+        var uriPath = path + '\\' + uri;
+        fs.readFile(uriPath, function (err, data) {
+            if (err) {
+                if (callback) {
```

Comment:
This was the only way I was able to get the jasmine test to recognize the error without the test stopping at the throw statement - I attempted to separate out the function that throws the exception so I could call it directly from the testing file as we discussed online, but I might have misinterpreted what you meant.